### PR TITLE
p5.Color.toString() now returns multiple string formats

### PR DESCRIPTION
**Pull request by stellartux**
Edited p5.Color.toString() to be able to format the string as RGBA, HSBA, HSLA, percentage values or hex color code.

### CODE REVIEW

**Comment by Spongman** [comment]
Maybe the 'hexN' versions should omit the '#' prefix? Then it would be easier to use this with other prefies (like '0x', '&', etc...)

**Comment by amoetodi** [comment]
Good idea, I'll make those changes.

**Comment by amoetodi** [comment]
This should be good to go now, unless you can think of anything else.

File: src/color/p5.Color.js [hunk 1]
```diff
@@ -52,8 +52,18 @@ p5.Color = function(renderer, vals) {
...
 /**
+ * This function returns the color formatted as a string. This can be useful
+ * for debugging, or for using p5.js with other libraries.
  * @method toString
- * @return {String}
+ * @param {Number|String} [format] How the color string will be formatted.
+ * Leaving this empty formats the string as rgba(r, g, b, a).
+ * '#rgb' '#rgba' '#rrggbb' and '#rrggbbaa' format as hexadecimal color codes.
+ * 'hex3' 'hex4' 'hex6' and 'hex8' return the same, without the '#''
+ * 16 'hex' and 'web-color' are aliases of 'hex6'.
```
Comment (lmccart) [inline src/color/p5.Color.js]: are these ones necessary? I am thinking of a beginner that reads this, it feels a bit overwhelming in the options and text to read through right now.

File: src/color/p5.Color.js [hunk 2]
```diff
@@ -52,8 +52,18 @@ p5.Color = function(renderer, vals) {
...
 
 /**
+ * This function returns the color formatted as a string. This can be useful
+ * for debugging, or for using p5.js with other libraries.
  * @method toString
- * @return {String}
+ * @param {Number|String} [format] How the color string will be formatted.
+ * Leaving this empty formats the string as rgba(r, g, b, a).
+ * '#rgb' '#rgba' '#rrggbb' and '#rrggbbaa' format as hexadecimal color codes.
+ * 'hex3' 'hex4' 'hex6' and 'hex8' return the same, without the '#''
```
Comment (lmccart) [inline src/color/p5.Color.js]: is this one necessary? it's adding in new language that we haven't really used elsewhere in the library. the other formats mostly align with inputs for the color functions and feel more intuitive. couldn't the user just use substring to remove the '#' if desired?

Comment (stellartux) [reply src/color/p5.Color.js]: I get where you're coming from, I was torn between trying to be easy to understand and trying to cover all the options someone might guess. Do you think it'd be better to remove the 'hex' options altogether, just have '#rrggbb' etc for the different hex color codes?

**Comment by lmccart** [comment]
thanks!

**Pull request merged**